doc/releases/changelog-dev.md

```diff
@@ -4,6 +4,72 @@
 
 <h3>New features since last release</h3>
 
+* Added new feature `qml.ops.op_math.Product.py` that was created using Sum.py as a template. This new 
+  feature allows for the creation of a product operation between operators. The product of rotation operatiors is 
```

Comment:
Hi @Miguelvs23 , I think there is some misconception here: Rotations are not commutative! Take for example XY = Z vs YX = -Z. Or a real world example: Take your phone laying on a table. rotate it 90º around the table's x axis, then 90º around the tables y axis, - then compare that with the reverse order (consistently all rotations clockwise) - your phone will end in very different positions.

Reply:
Maybe this is confused with the following observation, which is true: In order to generate a general rotation on the [Bloch sphere](https://en.wikipedia.org/wiki/Bloch_sphere) for a single qubit, it suffices to have two consecutive rotations around either two of the X, Y, and Z axes.

I.e. Any rotation RX(x1) RY(x2) can be written as RY(y1) RX(y2), for _different angles_ though! So for variational algorithms, it does indeed not matter in which order you apply the rotation gates. (Except for RZ rotations being redundant on the initial layer where qubits are initialized in |0> states, but this is a different story).

Reply:
I dont know what is your level of understanding of these concepts. If something is unclear, please dont hesitate to ask. There are no bad questions!

Reply:
Maybe the concept of commutativity was not clear to me, as I understood that the commutativity appears for rotation matrices in general. It is so that there is a general rotation when performing RX RY, so it would suffice to have two rotations for any general rotation of a qubit in the block sphere. 

The update allows for this scenario with different angles, so I think for the case for RX RY one can obtain a general rotation for the single qubit, then for more than 2 rotations of this type it is necessary to analyze the system. I will dig deeper into this subject, as I am still learning, but thank you very much!

In addition, I will consider the methods for hermiticity and commutativity employed in Sum.py, so that I can implement the same methods for Product.py. Thanks, @Qottmann

Reply:
Maybe the solution for the commutativity issue would be to demand that the order of the list of rotations be in the desired order, or something like that.

Reply:
I also thought of considering the general GL(2) case of matrices, for considering non pure states but I do not know if this is something useful or not

Reply:
The commutativity was just something I wanted to clarify since I saw it written here and in the doc string :) I just wrote a review on the main page with some general feedback on the implementation.